sky/resources.py

```diff
@@ -867,40 +867,40 @@ def from_yaml_config(cls, config: Optional[Dict[str, str]]) -> 'Resources':
         if config is None:
             return Resources()
 
-        backend_utils.validate_schema(config, schemas.get_resources_schema(),
-                                      'Invalid resources YAML: ')
+        backend_utils.validate_schema(config,
+                                      schemas.get_resources_schema(),
+                                      'Invalid resources YAML: ',
+                                      skip_none=True)
 
         resources_fields = {}
-        if config.get('cloud') is not None:
-            resources_fields['cloud'] = clouds.CLOUD_REGISTRY.from_str(
-                config.pop('cloud'))
-        if config.get('instance_type') is not None:
-            resources_fields['instance_type'] = config.pop('instance_type')
-        if config.get('cpus') is not None:
-            resources_fields['cpus'] = str(config.pop('cpus'))
-        if config.get('memory') is not None:
-            resources_fields['memory'] = str(config.pop('memory'))
-        if config.get('accelerators') is not None:
-            resources_fields['accelerators'] = config.pop('accelerators')
-        if config.get('accelerator_args') is not None:
+        resources_fields['cloud'] = clouds.CLOUD_REGISTRY.from_str(
+            config.pop('cloud', None))
```

Comment:
These changes may be harmless but change previous semantics. E.g., previously, `resources_fields['cloud']` doesn't get assigned if `cloud` field is not in `config`; but now it does.

Is it cleaner to revert the changes here somehow? Maybe make `validate_schema` remove None values? Note: this is just a question for discussion.

Reply:
Agreed that it is changes semantics. The reason why these changes are required is because the previous code had an implicit assumption that there are no `None` values in the code, so it depends if this something we want to avoid. Personally, I think that it is better to avoid that assumption and handle `None` explicitly, even if this may lead to semantic changes.